convert.py

```diff
@@ -38,27 +38,43 @@
 parser.add_argument('--num-train', type=int, default=19000000,
                     help='number of data for training')
 parser.add_argument('--no-negative', action='store_true', help="write no negative examples")
-parser.add_argument('--negative-num', type=int, default=99,
+parser.add_argument('--negative-num', type=int, default=999,
                     help='number of negatives per example')
 
 def get_movielens_data(data_dir, dataset):
     if not os.path.exists(data_dir + '%s.zip' % dataset):
         os.mkdir(data_dir)
         urllib.request.urlretrieve('http://files.grouplens.org/datasets/movielens/%s.zip' % dataset, data_dir + dataset + '.zip')
-    with zipfile.ZipFile(data_dir + "%s.zip" % dataset, "r") as f:
-        f.extractall(data_dir + "./")
+        with zipfile.ZipFile(data_dir + "%s.zip" % dataset, "r") as f:
+            f.extractall(data_dir + "./")
+
+def write_negative_examples(filename, val_data, max_items, negative_num=999):
+    f=open(filename,'a')
+    epoch=val_data.shape[0]//1000 +1 # number of blocks

```

Comment:
we can use `math.ceil(float(a)/float(b))` to avoid `val_data.shape[0]` can be divided by 1000.